### PR TITLE
Fibery Diaries Import

### PR DESCRIPTION
**Pull request by Drapegnik**
- `npm run init-articles`
- `npm run init-diaries`

https://github.com/babajka/babajka-frontend/pull/68

### CODE REVIEW

**Comment by Drapegnik** [comment]
[diaries fibery import](https://trello.com/c/1bTNqtmg/399-diaries-fibery-import)

File: src/api/specials/diary/controller.js [hunk 1]
```diff
@@ -1,59 +1,119 @@
...
+    const authors = await updateTags(tags, metadata, { skipValidation: true, skipMap: true });
+    const authorsIds = authors.reduce(
+      (acc, { fiberyId, _id }) => ({ ...acc, [fiberyId]: _id }),
+      {}
+    );
+    const diariesData = data.map(({ personality, day, month: { rank }, ...rest }) => ({
+      ...rest,
+      author: authorsIds[personality.fiberyId],
+      // TODO: check if rank is right
+      colloquialDateHash: (rank / ENUM_BASE + 1) * 100 + day,
```
Comment (UladBohdan) [inline src/api/specials/diary/controller.js]: is the comment explaining what's `colloquialDateHash` in model file still valid?
the `(month * 100 + day)` one

File: src/services/fibery/index.js [hunk 2]
```diff
@@ -73,12 +74,12 @@ const getArticleData = async url => {
     mapping: { Podcast: 'audio', publication: 'publishAt' },
     mapper: (key, lang = '') => (lang ? `locales.${lang}.${key}` : key),
     formatters: {
-      authors: TAG_FORMATER,
-      brands: TAG_FORMATER,
-      themes: TAG_FORMATER,
-      times: TAG_FORMATER,
-      personalities: TAG_FORMATER,
-      locations: TAG_FORMATER,
+      authors: TAGS_FORMATER,
```
Comment (UladBohdan) [inline src/services/fibery/index.js]: here and everywhere: formater is spelled as `formatter`...

File: src/services/fibery/index.js [hunk 3]
```diff
@@ -100,4 +101,35 @@ const getArticleData = async url => {
   return formatArticle(article);
 };
 
-export default { getArticleData };
+// TODO: add `month` input param (to reduce data amount)
```
Comment (UladBohdan) [inline src/services/fibery/index.js]: I don't think that's a problem.

File: src/utils/testing.js [hunk 4]
```diff
@@ -63,6 +64,44 @@ export const TEST_DATA = {
...
+            be: 'Уладзімір Караткевіч',
+          },
+          subtitle: {
+            be: '20 стг',
+          },
+          image: IMAGE_URL,
+          diaryImage: IMAGE_URL,
+          color: '#ff0000',
+          description: {
+            be: 'пісменнік',
```
Comment (UladBohdan) [inline src/utils/testing.js]: is that an "on purpose" typo ?

File: src/utils/testing.js [hunk 5]
```diff
@@ -63,6 +64,44 @@ export const TEST_DATA = {
...
+          },
+        },
+      },
+      kolas: {
+        fiberyId: 'testTag-kolas',
+        slug: 'kolas',
+        content: {
+          name: {
+            be: 'Якуб Клас',
```
Comment (UladBohdan) [inline src/utils/testing.js]: Колас

File: src/utils/testing.js [hunk 6]
```diff
@@ -63,6 +64,44 @@ export const TEST_DATA = {
...
+          },
+          subtitle: {
+            be: '20 стг',
+          },
+          image: IMAGE_URL,
+          diaryImage: IMAGE_URL,
+          color: '#ff0000',
+          description: {
+            be: 'пісменнік',
```
Comment (UladBohdan) [inline src/utils/testing.js]: пісьменнік

File: src/utils/testing.js (hunk 4 above)
Comment (UladBohdan) [reply src/utils/testing.js]: пісьменнік

File: src/utils/testing.js [hunk 7]
```diff
@@ -63,6 +64,44 @@ export const TEST_DATA = {
...
+          },
+        },
+      },
+    },
     brands: {
       default: {
         fiberyId: 'testTag-libra',
```
Comment (UladBohdan) [inline src/utils/testing.js]: libra is a new kurilka

File: src/api/specials/diary/controller.js [hunk 8]
```diff
@@ -1,59 +1,119 @@
...
   } catch (err) {
     return next(err);
   }
 };
 
-export default getDay;
+export const fiberyImport = async ({ user }, res, next) => {
+  try {
+    // FIXME: metadata.createdAt everytime new
+    const metadata = getInitObjectMetadata(user);
```
[the remark this answers is not in the document]
Comment (Drapegnik) [reply src/api/specials/diary/controller.js]: `importedAt`

File: src/db/scripts/importArticles.js [hunk 9]
```diff
@@ -0,0 +1,61 @@
+/* eslint-disable no-console */
+
+import connectDb from 'db';
+import { fiberyImport } from 'api/article/controller';
+
+import { retrieveMetadataTestingUser, mockRes } from './utils';
+
+const ARTICLES = [
```
[the remark this answers is not in the document]
Comment (Drapegnik) [reply src/db/scripts/importArticles.js]: yep

**Pull request merged**